src/dune_lang/package_dependency.ml

```diff
@@ -28,57 +28,68 @@ module Well_formed_name = struct
   module TT = Dune_util.Stringlike.Make (T)
 
   let decode =
-    let open Decoder in
+    let open Dune_sexp.Decoder in
     TT.decode >>| T.to_package_name
   ;;
 end
 
 let encode { name; constraint_ } =
-  let open Encoder in
+  let open Dune_sexp.Encoder in
   match constraint_ with
   | None -> Package_name.encode name
   | Some c -> pair Package_name.encode Package_constraint.encode (name, c)
 ;;
 
 (* Check for common typos in package dependency constraints *)
-let check_for_typo ~loc { name; constraint_ } =
-  let open Package_constraint in
-  match constraint_ with
-  | Some (Uop (Relop.Eq, Value.String_literal "version")) ->
-    let message =
-      User_message.make
-        ~loc
-        [ Pp.textf
-            "Possible typo in constraint for dependency %S: '(= version)' might be a \
-             mistake."
-            (Package_name.to_string name)
-        ]
-        ~hints:
+let dependency_constraint_variable_typo_warnings ~loc { name; constraint_ } =
+  (* Handle case where config system isn't initialized yet *)
+  let warnings_enabled =
+    try Dune_config.Config.(get typo_warnings) = `Enabled with
+    | Code_error.E _ -> true (* Only catches Code_error.E exceptions *)
+  in
+  (* Early return if warnings are disabled *)
```

Comment:
Nitpick about terminology, but this is not an early return. I don't think this comment adds information that isn't obvious reading the code.